### ISSUE TITLE
[7.0.2+TV] Cannot buy: Error at process._tickCallback

### ISSUE THREAD

**Issue by rene-re** [GuntharDeNiro/BTCT#377]
I am receiving trade signals through TV, trading on USDT-DASH pair. It worked several times.  However, now I am *always* receiving the following error:

2018/01/03 08:30:24: Checking prices...
2018/01/03 08:30:24: Received a BUY order...
[ { timestamp: 1514968223692,
    type: 'buy',
    exchange: 'bittrex',
    pair: 'USDT-DASH' } ]
Error: [object Object]
    at errorHandler (/snapshot/gunbot-build/ctx-wrapper.js:0:0)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:169:7)
undefined
undefined

What I already checked:
- Balance on Bittrex
- Other pairs
- Enabling debug option (however, no additional output is generated)

**Comment by rene-re** [comment]
Additional info: When I wrote "It worked several times", I remember it working with 7.0.2 RC3 but not with the final version.
I switched back to RC3 and now orders seem to be working again!

**Comment by vcyu08** [comment]
I'm getting a similar error and have not been able to complete a buy order with Gunbot XT Edition, 7.0.2.  Have sent a message to Gunthar via gunthy.org regarding this issue.

Below is from the log:

GBXT_v7.0.2 ETH-XLM Round # 4
------------------------------------------------------------------------------------------------
 ɢՍͶ฿Ǿͳ XT Edition  Scanning bittrex for trading opportunities 2018/01/03 20:04:02
------------------------------------------------------------------------------------------------
Last trade P/L for ETH-XLM was: 0.000 %
************************************************************************************************
Waiting to trade...Safety Switch is on...
************************************************************************************************

GBXT_v7.0.2 ETH-XLM Round # 5
------------------------------------------------------------------------------------------------
 ɢՍҊɃøͳ XT Edition  Scanning bittrex for trading opportunities 2018/01/03 20:04:16
------------------------------------------------------------------------------------------------
Last trade P/L for ETH-XLM was: 0.000 %
Error: [object Object]
    at errorHandler (C:\snapshot\Gunbot-master\ctx-wrapper.js:0:0)
    at 
    at process._tickCallback (internal/process/next_tick.js:169:7)

**Comment by vcyu08** [comment]
The exchange I'm using is also bittrex.

**Comment by monochromat13** [comment]
On Bittrex, receiving a similar error: 

GBXT_v7.0.1ETH-QTUM Round # 14
------------------------------------------------------------------------------------------------
 ɢƱͶɮ۝Ł XT Edition  Scanning bittrex for trading opportunities 2018/01/04 08:27:21
------------------------------------------------------------------------------------------------
Last trade P/L for ETH-QTUM was: 0.000 %
┌───────────────────┬────────────────┬────────────┬─────────────┬────────────┬─────────────┐
│ ɢƱÑƁøͳ XT Edition │ Stop limit hit │ Can we buy │ Can we sell │ Panic sell │ Can average │
├───────────────────┼────────────────┼────────────┼─────────────┼────────────┼─────────────┤
│ Core checks       │       NO       │     YES    │      NO     │     NO     │      NO     │
└───────────────────┴────────────────┴────────────┴─────────────┴────────────┴─────────────┘
2018/01/04 08:27:31: Last Bid price ETH-QTUM: 0.05158222 | Last Ask: 0.05199803
2018/01/04 08:27:31: Balance ETH: XXXXXX
2018/01/04 08:27:31: Balance QTUM: 0
2018/01/04 08:27:31: On Orders QTUM: 0
Processing strategy emotionless for ETH-QTUM on bittrex
BUY CYCLE
2018/01/04 08:27:31: Entering buy cycle
Error: INVALID_PERMISSION
    at errorHandler (/snapshot/gunbot-build/ctx-wrapper.js:0:0)
    at 
    at process._tickCallback (internal/process/next_tick.js:169:7)
[GUNBOT XT] No orders to recalculate

**Comment by monochromat13** [comment]
Also, @rene-re where did you find 7.02 RC3? Would like to revert if possible.

**Comment by rene-re** [comment]
@monochromat13 You are likely to have a API permission error. Quote:
> Error: INVALID_PERMISSION
The API key might lack the necessary permissions to do a trade.
The RC is available in the Telegram group.

**Comment by monochromat13** [comment]
Thanks @rene-re, that API key had limit but not market permissions. That might solve the problem. Cheers.

**Comment by vcyu08** [comment]
I think I figured out what the issue is.  I tried using Gunbot 7.0.1, and got the error message, Error: MIN_TRADE_REQUIREMENT_NOT_MET.  After doing some research, it looked to me that MIN_TRADE_REQUIREMENT_NOT_MET means that the buy order didn't meet Bittrex's minimum trading requirement of 0.001 BTC.  

Each Gunbot strategy has the parameter TRADING_LIMIT with default value 0.001.  So if I'm trading ETH-XLM pair, and the price of Lumen in Ethereum is 0.00094 ETH/XLM, then Gunbot could buy at most (0.001 ETH) / (0.00094 ETH/XLM), which is around 1.063830 XLM.  Given the price of of Lumen in Bitcoin is 0.00005910 BTC/XLM, 1.063830 XLM is about (0.00005910 BTC/XLM) * (1.063830 XLM) = 0.000062872 BTC, which is less than Bittrex's minimum trade requirement of 0.001 BTC.


I changed the value of the TRADING_LIMIT parameter from 0.001 to 0.15 for bbstepgain strategy.  In this case, Gunbot could buy at most (0.15 ETH) / (0.00094 ETH/XLM), which is around 159.57447 XML, which is about (0.00005910 BTC/XLM) * (159.57447 XML) ~ 0.009431 BTC, greater than Bittrex's minimum trading requirement of 0.001 BTC.

After the TRADING_LIMIT value was updated, Gunbot was able to place a buy order, and the order was completed in Bittrex.  Thanks!

Error: MIN_TRADE_REQUIREMENT_NOT_MET
    at errorHandler (C:\snapshot\Gunbot-master\ctx-wrapper.js:0:0)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:169:7)

**Comment by rene-re** [comment]
Happy to hear that you solved your issue. However, the issue I mentioned above seems to be similar but is not identical.

You received the error
> Error: MIN_TRADE_REQUIREMENT_NOT_MET

However, this bug report is about 
> Error: [object Object]

**Comment by vcyu08** [comment]
Thank you!  I think an issue with 7.0.2 is that the error message is not descriptive.  I started with 7.0.2 and got the error message Error: [object Object].  Then I tried 7.0.1, and the error message in 7.0.1 is Error: MIN_TRADE_REQUIREMENT_NOT_MET, which gave me some clue to work with.

**Comment by COSMIC-COCO** [comment]
I had this exact error message (for all coins):

Received a BUY order...
[ { timestamp: 1514968223692,
type: 'buy',
exchange: 'bittrex',
pair: 'USDT-DASH' } ]
Error: [object Object]
at errorHandler (/snapshot/gunbot-build/ctx-wrapper.js:0:0)
at 
at process._tickCallback (internal/process/next_tick.js:169:7)
undefined
undefined


I was able to resolve the error by increasing the trade limit for USDT to:

        "TV_TRADING_LIMIT_BUY": 25.0,
        "TV_PYRAMID": false,
        "TV_TRADING_LIMIT_SELL": 25.0,

I think this version of GB may be missing some error descriptions. The default trade size of 0.001 is too low for USDT.  5.0 was also too low. I didn't try anything in between 5 and 25. 

https://support.bittrex.com/hc/en-us/articles/115003004171-What-are-my-trade-limits-

**Comment by bhallouin** [comment]
I have the exact same issue with "Entering buy cycle". That happened when I launched the bot for the first time yesterday with pairs like BTC-ETH or BTC-BCC. First I thought I didn't give enough rights to my _Master Key_. I changed it in Bittrex config. and it started to work properly.

That happened again when I added few hours ago two new pairs (BTC-XML and BTC-SC) and reloaded the config. I didn't change my _Master Key_ configuration in the meantime... Can't find out how to solve this yet.

One thing which might be related : I have that **NaN** strange info. here for both pairs I just mentioned:

`2018/01/07 01:24:15: Got Trading History
2018/01/07 01:24:16: Checking Open Orders
Last trade P/L for BTC-XLM was: 0.000 %
2018/01/07 01:24:16: Price to buy: 0.000042065037855
2018/01/07 01:24:16: Last Bid/Strategy Exit Point: 0.00004240/**NaN**
2018/01/07 01:24:16: Last Ask/Strategy Entry Point: 0.00004244/0.000042065037855
2018/01/07 01:24:16: XTrend for BTC-XLM:    DOWN!!!`

By the way, the bot is selling properly, so I think this is not an issue with the API key.
